### PR TITLE
Add multiprocess neccessary function

### PR DESCRIPTION
**Pull request by younger15**

### CODE REVIEW

**Review by Jazz86 — APPROVED** [review]
approve.

File: src/Header/MainManager.h [hunk 1]
```diff
@@ -19,16 +19,26 @@ class MainManager {
  public:
   // httpListener callback, do fork and new requestHandler to call api, child
   // process add to parents vector for managing.
-  void NewRequest(const std::string &request_str);
+  void NewRequest(const std::string &request_fd);
```
Comment (john0312) [inline src/Header/MainManager.h]: std::string is probably not an fd.

File: src/Header/MainManager.h [hunk 2]
```diff
@@ -37,12 +47,16 @@ class MainManager {
   // Delete assignment operator
   MainManager &operator=(const MainManager &cpy) = delete;
 
+  protected:
+  // singleton code convention method.
```
Comment (john0312) [inline src/Header/MainManager.h:53]: Please clang-format.

**Review by john0312 — APPROVED** [review]
Please fix the items in the review before merging. Thank you.

File: src/Header/MainManager.h (hunk 2 above)
Comment (younger15) [reply src/Header/MainManager.h:53]: Done

File: src/Header/MainManager.h (hunk 1 above)
Comment (younger15) [reply src/Header/MainManager.h]: Done

**Pull request merged**